test/unit/argument_parser/format_parse_test.cpp

```diff
@@ -936,3 +926,37 @@ TEST(parse_test, subcommand_argument_parser_success)
                      seqan3::argument_parser_error);
     }
 }
+
+TEST(parse_test, issue1544)
+{
+    {   // unknown option that has a prefix of a known option
+        std::string option_value;
+        const char * argv[] = {"./argument_parser_test", "--foo", "hallo", "--foo-bar", "ballo"};
+        seqan3::argument_parser parser{"test_parser", 5, argv, false};
+        parser.add_option(option_value, 'f', "foo", "this is a string option.");
+
+        EXPECT_THROW(parser.parse(), seqan3::unknown_option);
+    }
+
+    {   // known option that has a prefix of a unknown option
+        std::string option_value;
+        const char * argv[] = {"./argument_parser_test", "--foo", "hallo", "--foo-bar", "ballo"};
+        seqan3::argument_parser parser{"test_parser", 5, argv, false};
+        parser.add_option(option_value, 'f', "foo-bar", "this is a string option.");
+
+        EXPECT_THROW(parser.parse(), seqan3::unknown_option);
+    }
+
+    {   // known option is a prefix of another known option
+        std::string foo_option_value;
+        std::string foobar_option_value;
+        const char * argv[] = {"./argument_parser_test", "--foo", "hallo", "--foo-bar", "ballo"};
+        seqan3::argument_parser parser{"test_parser", 5, argv, false};
+        parser.add_option(foo_option_value, 'f', "foo", "this is a prefix of foobar.");
+        parser.add_option(foobar_option_value, 'b', "foo-bar", "this has prefix foo.");
+
+        EXPECT_NO_THROW(parser.parse());
+        EXPECT_EQ(foo_option_value, "hallo");
+        EXPECT_EQ(foobar_option_value, "ballo");
+    }
+}
```

Comment:
Is there a test missing for calling the argument parser with: `--foohallo`. So without the equal sign or separated with a space? This should than throw something, right?

Reply:
I think it is tested by option `--foo` with value `--foo-bar` with the value being `-bar`? I can remove the dash between `foo-bar` if this makes it clearer?

Reply:
Ah ok, from the test design I thought that this was a different thing to test. But it does test the same that is right. I guess the issue is not the dash inbetween but that the tests checks if the one is a prefix of another. So from the test I don't actually see that long-option value pairs without a separation are not allowed. 
```cpp
    {   // wrong separation of long value:
        std::string option_value;
        const char * argv[] = {"./argument_parser_test", "--foohallo"};
        seqan3::argument_parser parser{"test_parser", 2, argv, false};
        parser.add_option(option_value, 'f', "foo", "this is a string option.");

        EXPECT_THROW(parser.parse(), seqan3::unknown_option);
    }
```

Reply:
Alright. I added the test next to the existing ones. Doesn't hurt to be expressive although they internally test the same thing